### PR TITLE
Guard against various Faraday exception response formats

### PR DESCRIPTION
**Pull request by dalibor**
## What does this pull request do?

- Guard against other Faraday exception response formats
- Fix randomly failing test by stubbing getconf shell out

## Why is it important?

To prevent errors while fetching `status` in Faraday rescue client/server errors block.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [x] My code follows the style guidelines of this project (See `.rubocop.yml`)
- [x] I have rebased my changes on top of the latest main branch

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-agent-ruby/blob/main/CONTRIBUTING.md#testing) pass locally with my changes

- [x] I have made corresponding changes to the documentation
- [x] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
- [x] I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
- [x] Added an API method or config option? Document in which version this will be introduced

### CODE REVIEW

File: lib/elastic_apm/spies/faraday.rb [hunk 1]
```diff
@@ -109,8 +109,14 @@ def run_request(method, url, body, headers, &block)
                   yield req if block
                 end
               rescue Faraday::ClientError, Faraday::ServerError => e # Faraday::Response::RaiseError
-                status = e.response_status if e.respond_to?(:response_status)
-                status ||= e.response&.fetch(:status)
```
Comment (dalibor) [inline lib/elastic_apm/spies/faraday.rb:113]: Not sure when `e.response` would be a hash, but I've preserved this behavior down below and I'm also handling a _possible_ hash response without a `:status` key as well.

The primary error I want to fix is:

```
undefined method `fetch' for #<Faraday::Response:0x000000000c657078>
```

**Comment by dalibor** [comment]
@estolfo @picandocodigo Could someone from the team review this PR please?

Also, we'll need a new official version released since [4.7.0](https://rubygems.org/gems/elastic-apm/versions/4.7.0) is not compatible with Rails 7.1.

Thanks!

**Comment by picandocodigo** [comment]
@elasticmachine run elasticsearch-ci/docs

File: lib/elastic_apm/spies/faraday.rb (hunk 1 above)
Comment (picandocodigo) [reply lib/elastic_apm/spies/faraday.rb:113]: > Not sure when e.response would be a hash

This applies in older versions of Faraday, you can see the change [here](https://github.com/elastic/apm-agent-ruby/pull/1419) and there's more information following the corresponding issue.

**Review by picandocodigo — APPROVED** [review]

**Pull request merged**